### ISSUE TITLE
registry_auth::token_not_logged is failing randomly

### ISSUE THREAD

**Issue by ehuss** [rust-lang/cargo#12639]
The test `registry_auth::token_not_logged` has been failing in CI randomly, and I don't immediately have thoughts why. 

* https://github.com/rust-lang/cargo/pull/12624#issuecomment-1707712539
* https://github.com/rust-lang/cargo/pull/12575#issuecomment-1697926451
* https://github.com/rust-lang/cargo/pull/12317#issuecomment-1611319929
* https://github.com/rust-lang/cargo/pull/12539#issuecomment-1688209161
* https://github.com/rust-lang/cargo/pull/12632#issuecomment-1709381422

```
---- registry_auth::token_not_logged stdout ----
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: None,
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/3/b/bar",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/dl/bar/1.0.0/download",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
/home/runner/work/cargo/cargo/target/tmp/cit/t2285/dl/bar/1.0.0/download
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/api/v1/crates/new",
        query: None,
        fragment: None,
    },
    method: "put",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            37477,
        ),
        path: "/index/3/f/foo",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
thread 'registry_auth::token_not_logged' panicked at tests/testsuite/registry_auth.rs:499:5:
assertion `left == right` failed
  left: 8
 right: 7
```

This issue has been assigned to @hi-rustin via [this comment](https://github.com/rust-lang/cargo/issues/12639#issuecomment-1709500745).

**Comment by ehuss** [comment]
@rustbot assign @hi-rustin 

@hi-rustin indicated they may be able to look at this at https://github.com/rust-lang/cargo/pull/12624#issuecomment-1707756899.  Let me know if you have any questions or want me to take a look to give you any pointers.

**Comment by ehuss** [comment]
@hi-rustin I'm not sure if you've had a chance to look at this, but since it has been a little disruptive I went ahead and posted #12726 and #12727 to fix this. There seemed to be two independent errors, hence the two PRs.

**Comment by ehuss** [comment]
Closing as this should now be resolved by #12726 and #12727.
